### PR TITLE
Add v0 cid support

### PR DESCRIPTION
**Pull request by storming0x**
## Description
We need to expose the v0 Hash as well for clients to choose.

https://app.clubhouse.io/terminalsystems/story/15902/make-s3x-hashes-v0-compatible

## Motivation and Context
currently we were only exposing the v1 hash

## How to test this PR?
run gateway-s3x-utils-hash_test.go

Screen Shot 2020-05-15 at 1 52 26 PM

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: cmd/gateway/s3x/gateway-s3x-bucket_test.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"testing"
 	"time"
 
-	minio "github.com/RTradeLtd/s3x/cmd"
+	minio "github.com/minio/minio/cmd"
```
Comment (jsonsivar) [inline cmd/gateway/s3x/gateway-s3x-bucket_test.go:8]: Good catch

File: cmd/gateway/s3x/gateway-s3x-bucket_test.go [hunk 2]
```diff
@@ -123,7 +124,7 @@ func testS3XBucket(t *testing.T, dsType DSType) {
 		}
 		for _, tt := range tests {
 			t.Run(tt.name, func(t *testing.T) {
-				err := gateway.DeleteBucket(ctx, tt.args.bucketName)
+				err := gateway.DeleteBucket(ctx, tt.args.bucketName, false)
```
Comment (jsonsivar) [inline cmd/gateway/s3x/gateway-s3x-bucket_test.go:127]: I thought this was already in `master`. Maybe just a github issue

File: cmd/gateway/s3x/gateway-s3x-ledger-store.go [hunk 3]
```diff
@@ -41,16 +40,20 @@ type ledgerStore struct {
 	pmapLocker sync.Mutex   //a lock to protect the l.MultipartUploads map from concurrent access
 
 	cleanup []func() error //a list of functions to call before we close the backing database.
+
+	oh OperationHelper // hook operations that can be called after an operation is finished
 }
 
 func newLedgerStore(ds datastore.Batching, dag pb.NodeAPIClient) (*ledgerStore, error) {
+	oh := NewOperationHelper(nil)
```
Comment (jsonsivar) [inline cmd/gateway/s3x/gateway-s3x-ledger-store.go:48]: Very nice refactor thank you.

**Review by jsonsivar — APPROVED** [review]
LGTM

File: cmd/gateway/s3x/gateway-s3x-bucket_test.go (hunk 2 above)
Comment (storming0x) [reply cmd/gateway/s3x/gateway-s3x-bucket_test.go:127]: ahh it is, but i did a squash merge so the diff shows it

**Pull request merged**